docs/source/serialization.mdx

```diff
@@ -49,257 +60,365 @@ Ready-made configurations include the following models:
-reason, when using `convert_graph_to_onnx.py` optimizations are not enabled, ensuring the model can be easily
-exported to various hardware. Optimizations can then be enabled when loading the model through ONNX runtime for
-inference.
+Every configuration object must implement the `inputs` property and return an
+`OrderedDict`, where each key corresponds to an expected input, and each value
+indicates the axis of that input. The `OrderedDict` is needed to ensure the
+inputs are matched against their relative position within the
+`PreTrainedModel.forward()` method when tracing the graph.
 
+For DistilBERT, we can see that two inputs are required: `input_ids` and
+`attention_mask`. These inputs have the same shape of `(batch_size,
+sequence_length)` which is why we see the same axes used in the configuration.
 
-<Tip>
+Once you have implemented an ONNX configuration, you can instantiate it by
+providing the base model's configuration as follows:
 
-When quantization is enabled (see below), `convert_graph_to_onnx.py` script will enable optimizations on the
-model because quantization would modify the underlying graph making it impossible for ONNX runtime to do the
-optimizations afterwards.
+```python
+from transformers import AutoConfig
 
-</Tip>
+config = AutoConfig.from_pretrained("distilbert-base-uncased")
+onnx_config = DistilBertOnnxConfig(config)
+```
 
-<Tip>
+The resulting object has several useful properties. For example you can view the
+ONNX operator set that will be used during the export:
 
-For more information about the optimizations enabled by ONNXRuntime, please have a look at the [ONNXRuntime Github](https://github.com/microsoft/onnxruntime/tree/master/onnxruntime/python/tools/transformers).
+```python
+# Returns 11
+print(onnx_config.default_onnx_opset)
+```
 
-</Tip>
+You can also view the outputs associated with the model as follows:
 
-### Quantization
+```python
+print(onnx_config.outputs)
+```
 
-ONNX exporter supports generating a quantized version of the model to allow efficient inference.
+which gives:
 
-Quantization works by converting the memory representation of the parameters in the neural network to a compact integer
-format. By default, weights of a neural network are stored as single-precision float (*float32*) which can express a
-wide-range of floating-point numbers with decent precision. These properties are especially interesting at training
-where you want fine-grained representation.
+```python
+OrderedDict([("last_hidden_state", {0: "batch", 1: "sequence"})])
+```
 
-On the other hand, after the training phase, it has been shown one can greatly reduce the range and the precision of
-*float32* numbers without changing the performances of the neural network.
+Notice that the outputs property follows the same structure as the inputs; it
+returns an `OrderedDict` of named outputs and their shapes. The output structure
```

Comment:
It is actually not always an `OrderedDict` I think, it depends on the configuration (as we did not always make sure to use that), but the important takeaway is that the inputs and outputs must be fed in the proper order to the export function.

Reply:
Thanks for catching this! I've now replaced `OrderedDict` with "mapping" in this sentence and added a tip box below to clarify the role of the `OrderedDict` in this particular example.